### PR TITLE
Codacy 

### PR DESCRIPTION
**Pull request by AndyScherzinger**
This yet another housekeeping PR which fixes several codacy warning categories, malicious findbugs warnings and some java compiler warnings:

* :mag: codacy: this statement should have braces
* :mag: codacy: a class which only has private constructors should be final
* :mag: codacy: position literals first in String comparisons for EqualsIgnoreCase
* :mag: codacy: replace with local variable
* :mag: codacy: combine nested if statements
* :mag: codacy: make utilities classes final with private constructor 
* :mag: codacy: avoid using Literals in Conditional Statements
* :bug: findbugs: malicious code vulnerability Warnings (4, missing file stream closing in finally block)

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Finally this PR is ready to review 🎉 
@tobiasKaminsky

File: src/main/java/com/owncloud/android/operations/UploadFileOperation.java [hunk 1]
```diff
@@ -699,12 +699,10 @@ private RemoteOperationResult encryptedUpload(OwnCloudClient client, OCFile pare
 
     private RemoteOperationResult unlockFolder(OCFile parentFolder, OwnCloudClient client, String token) {
         if (token != null) {
-            UnlockFileOperation unlockFileOperation = new UnlockFileOperation(parentFolder.getLocalId(), token);
-            RemoteOperationResult unlockFileOperationResult = unlockFileOperation.execute(client, true);
-
-            return unlockFileOperationResult;
-        } else
+            return new UnlockFileOperation(parentFolder.getLocalId(), token).execute(client, true);
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/operations/UploadFileOperation.java]: I really like this condensed way :+1:

File: src/main/java/com/owncloud/android/utils/DisplayUtils.java [hunk 2]
```diff
@@ -113,6 +113,10 @@
     private static final String HTTP_PROTOCOL = "http://";
     private static final String HTTPS_PROTOCOL = "https://";
     private static final String TWITTER_HANDLE_PREFIX = "@";
+    private static final int MIMETYPE_PARTS_COUNT = 2;
+    private static final int BYTE_SIZE_DIVIDER = 1024;
+    private static final double BYTE_SIZE_DIVIDER_DOUBLE = 1024.;
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/utils/DisplayUtils.java]: Either 1024.0 or 1024d or simply 1024 (latest should also work according to javarepl).
The dot is a bit confusing.

**Comment by tobiasKaminsky** [comment]
:+1:

File: src/main/java/com/owncloud/android/operations/UploadFileOperation.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/operations/UploadFileOperation.java]: :heart:

File: src/main/java/com/owncloud/android/utils/DisplayUtils.java (hunk 2 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/utils/DisplayUtils.java]: used 1024.0 via https://github.com/nextcloud/android/pull/3010/commits/ee4ec654dec53b3d46e650d8b3585501f3cf53bd

**Comment by AndyScherzinger** [comment]
Just waiting for CI and codacy, fixed the code review comment.

**Pull request merged**